controllers/databasecluster_controller.go

```diff
@@ -545,7 +545,6 @@ func (r *DatabaseClusterReconciler) reconcilePSMDB(ctx context.Context, req ctrl
 		psmdb.Spec.Secrets = &psmdbv1.SecretsSpec{
 			Users: database.Spec.Engine.UserSecretsName,
 		}
-		psmdb.Spec.Mongod.Security.EncryptionKeySecret = fmt.Sprintf("%s-mongodb-encryption-key", database.Name)
```

Comment:
We need to move it to `spec.secrets.encryptionKey` please see https://github.com/percona/percona-server-mongodb-operator/blob/main/deploy/cr.yaml#L40